### ISSUE TITLE
Action output type ignoring nullability of fields and relationships

### ISSUE THREAD

**Issue by GormanFletcher** [hasura/graphql-engine#6873]
I'm declaring an action like so:

![image](https://user-images.githubusercontent.com/78770467/117158191-340a0d00-ad8d-11eb-9a3b-758313709464.png)

The action handler returns type `{userId: string | null}`. I've added a relationship, `user`, to this action, mapping the returned `userId` to my `users` DB table.

Hasura is generating types that assert the `userId` and `user` are non-nullable. The GraphiQL console in Hasura is happy to handle the null values in my test queries, but my frontend blows up at the schema violation.

Here's what the Explorer in Hasura shows for the types:
![image](https://user-images.githubusercontent.com/78770467/117157241-6ebf7580-ad8c-11eb-9d45-ec32288d92d0.png)

Here's the query I'm making to Hasura:

```graphql
query FetchUser {
  currentUserId {
    user {
      email
      name
      userId
    }
  }
}     
```

Hasura responds with this error:

```json
{
  "errors": [
    {
      "extensions": {
        "path": "$.selectionSet.currentUserId.selectionSet.user",
        "code": "validation-failed"
      },
      "message": "field \"user\" not found in type: 'CurrentUserIdOutput'"
    }
  ]
}
```

I'm using Hasura v2.0.0-alpha.10.

**Comment by GormanFletcher** [comment]
It looks like this is a duplicate of #4405.

**Comment by tirumaraiselvan** [comment]
Dupe of #4405.

This has been fixed v2.5 onwards. Action object relationships are all nullable now.
